server/build.go

```diff
@@ -248,13 +248,30 @@ func PostBuild(c *gin.Context) {
 		build.Deploy = c.DefaultQuery("deploy_to", build.Deploy)
 	}
 
+	// Read query string parameters into buildParams, exclude reservedParams
+	var buildParams = map[string]string{}
+	reservedParams := []string{"fork", "event", "deply_to"}
+	for key, val := range c.Request.URL.Query() {
+		// We only accept string literals, because build parameters will be
+		// injected as environment variables
+		buildParams[key] = val[0]
+		for _, r := range reservedParams {
+			if r == key {
+				delete(buildParams, r)
+			}
+		}
+	}
+
```

Comment:
we could probably slightly alter the code to avoid the extra allocation of the map with a global variable, and avoid extra operations adding and removing items from the map as well with lookup:

```
	var buildParams = map[string]string{}
	for key, val := range c.Request.URL.Query() {
		if _, ok := reservedParams[key]; !ok {
			buildParams[key]=val
		}
	}
```

or a simple switch statement:

```
	var buildParams = map[string]string{}
	for key, val := range c.Request.URL.Query() {
		switch {
                case "fork", "event", "deply_to":
		default:
			buildParams[key]=val
		}
	}
```